### PR TITLE
maplibre support for Qt6 and MSVC 2019

### PR DESCRIPTION
**Pull request by dpaulat**
Update maplibre-gl-native to compile using Qt 6.1.1 and MSVC 2019

@ntadej should probably have his PR merged first:
https://github.com/maplibre/maplibre-gl-native/pull/118

Once his is merged, I'll work to resolve any conflicts.

### CODE REVIEW

**Comment by dpaulat** [comment]
Open issues are:
- https://github.com/maplibre/maplibre-gl-native/issues/124
- Missing zlib dependency, currently must be resolved externally

**Comment by ntadej** [comment]
Thanks @dpaulat!

I have also been playing with this a bit and I have some additional fixes prepared, namely:
 - One can use icu support and zlib from Qt
 - I did some cleanup of the CMake configuration (but can wait for this MR to go in)
 - I fixed various warnings

**Comment by dpaulat** [comment]
With #118 merged, I rebased and resolved conflicts.  This PR is ready for review.

**Review by ntadej — CHANGES_REQUESTED** [review]

File: platform/qt/qt.cmake
Comment (dpaulat) [reply platform/qt/qt.cmake:7]: Just these specific components?  I'd need to do some digging to figure out what this should be.  If you have an idea of what this should be, I can leave this to you.

https://doc.qt.io/qt-5/cmake-get-started.html

**Comment by klokan** [comment]
This looks like a great activity @dpaulat & @ntadej. 👍  Thank you very much for looking into this.

Is there a first demo app? Or do you develop that for some particular project of yours?
 
@petr-pokorny-1 Could you please check the code and provide your comments in a review - once you are back from holiday?
 
BTW We could then eventually update https://www.maptiler.com/news/2019/04/using-maptiler-maps-inside-qt/ or reuse the older sample app at https://github.com/maptiler/maptiler-qml-demo to demo the functionality - esp. if there is no other demo app.

**Comment by dpaulat** [comment]
Thank you!  There is a subproject in this repository called mbgl-qt that provides demo functionality.  I can post a screenshot this evening.

**Comment by ntadej** [comment]
> Is there a first demo app? Or do you develop that for some particular project of yours?

I have a hobby project where I already had to patch existing Mapbox implementation and having something standalone and independent from Qt releases would be much more useful.

Also I'm happy to develop the revised version of QML plugin under the MapLibre umbrella. I have a prototype ready but I'm still cleaning up and streamlining the build system (some more MRs incoming). https://github.com/ntadej/qt-geoservices-maplibre

**Comment by dpaulat** [comment]
> Thank you! There is a subproject in this repository called mbgl-qt that provides demo functionality. I can post a screenshot this evening.

2021-07-22 21_53_30-Mapbox GL_ Basic

**Comment by petr-pokorny-1** [comment]
@dpaulat it looks like your changes are causing Android and iOS SDKs fails to compile. It is mostly due to changes you made in constructor/methods signatures. Do you have Android/iOS dev environment to fix these issues?

**Comment by ntadej** [comment]
> @dpaulat it looks like your changes are causing Android and iOS SDKs fail to compile. It is mostly due to changes you made in constructor/methods signatures. Do you have Android/iOS dev environment to fix these issues?

I think this commit should fix it: https://github.com/ntadej/maplibre-gl-native/commit/5bedf45eceecfd650046a555f852dae1311f0fcb

**Comment by dpaulat** [comment]
I will work to spin up an Android environment and see what happened. Hopefully the Android and iOS errors are the same, since I don't have a suitable environment for that.
________________________________

**Comment by dpaulat** [comment]
Even better. I will get this cherry picked when I get back to my computer.
________________________________

**Comment by petr-pokorny-1** [comment]
@ntadej @dpaulat Great - thanks guys!

**Review by petr-pokorny-1 — APPROVED** [review]
Looks good to me. A lot of work, thanks!

**Comment by petr-pokorny-1** [comment]
Any chance to write CI workflow for QT and Windows (similar to those in /.github/workflows) ?

**Comment by ntadej** [comment]
I have something prepared for Mac, android and windows. Plan to open a MR when my other changes will be merged.

I will be on vacation now for two weeks so after this one is merged, I will probably just open one or two MRs.

**Comment by dpaulat** [comment]
Any other actions for this PR prior to merge?  Thanks!

**Comment by roblabs** [comment]
This was a lot of work!  Thanks for contributing to MapLibre & FOSS4G.  Looking forward to the CI workflows for GitHub as well.   I reviewed the iOS build for extra warnings (there were none).

**Pull request merged**
